### ISSUE TITLE
Parallel Deploys

### ISSUE THREAD

**Issue by lovesegfault** [serokell/deploy-rs#46]
Would you consider allowing deploy-rs to deploy to multiple nodes simultaneously?

I wanted to attempt implementing that, but figured I'd check before so as not to end up wasting time.

**Comment by notgne2** [comment]
Absolutely, but one thing to bare in mind is I would one day like to support deploying multiple profiles to the same node, which will require doing away with our `profilesOrder` attribute in favor of a dependency map.

I'm not suggesting you go through that hard work too (though I would obviously appreciate it), but if it ends up being relevant, if you could make your logic not too hard to develop into supporting paralell profile deployment that would make it easier for me in the future.

**Comment by nrdxp** [comment]
I've been playing with tokio threads lately so if the author would still like to start on this or has already please share and I'll try to help :smile:

This would really be an infinitely scalable solution for large deployments.

**Comment by jzbor** [comment]
I think a nice idea would be building in parallel (as this is the most work-intensive part) and deploying in serial to ensure low down times and some degree of determinism.
